### ISSUE TITLE
Error when including a library from my RAML spec

### ISSUE THREAD

**Issue by KevinMitchell** [mulesoft/api-workbench#397]
I define my RAML file in a directory, and create subdirectories called traits and libraries.   Inside the traits folder I define a collection of traits, and then in the libraries folder I define common.raml as follows:

    #%RAML 1.0 Library

    traits:
      pageable:   !include ../traits/pageable.raml
      ...

The API workbench seems happy with all this.  In my main RAML spec I then declare

    uses:
      common: !include libraries/common.raml

This fails with
      `Error Unknown node: 'traits'`
      `Trace Unknown node 'traits' at line 3 col 1 in libraries/common.raml`

It's not clear to me whether I'm doing something wrong or there's a bug.

Kevin

**Comment by jarrodek** [comment]
I have the same problem.
I've created simplest API example:

File location: `schemas/test-library.raml`
```
#%RAML 1.0 Library

usage: |
  Test library

types:
  File:
    properties:
      name:
      length:
        type: integer
```
And main file: `api.raml`
```
#%RAML 1.0
title: New API

uses: 
  test: !include schemas/test-library.raml
```

I have an error in `api.raml` file  on line 4: 
> Error Unknown node: 'types'
Trace Unknown node 'types' at line 3 col 1 in schemas/test-library.raml

I'm lost here and I'm not sure how to fix this.

**Comment by jarrodek** [comment]
OK, got it.
Libraries are included without `!include` keyword. It should look like this:
```
uses: 
  test: schemas/test-library.raml
```
and then it will work.

Though, the parser should point this out instead of throwing and error that means nothing.

**Comment by KevinMitchell** [comment]
Thanks, well spotted.  I'll try that.
Kevin

**Comment by ddenisenko** [comment]
https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md#libraries - please check this chapter to see how to use libraries.

**Comment by KevinMitchell** [comment]
I think the issue here is that we have to use !include to reference a typed fragment, but not use !include when including a library.  That's fine, but a common error is then likely to be someone accidentally using !include with a library, and the error you get is then non-obvious.  So explicitly checking for this, and returning a more explicit error, would help here.

**Comment by ddenisenko** [comment]
I agree, catching a user when the library is included as a fragment and displaying a warning in this case is a good idea.

**Comment by jarrodek** [comment]
@gis1330 please, use code formatting for everyone's well being ;)

`MyType`  is a reference to a library, not a type. My example above is not correct.
Your declaration should look something like this:

```yaml
uses:
  MyLib: <RAML_path>
/:
  post:
    body:
      application/json:
        type: MyLib.MyType
```

**Comment by gis1330** [comment]
I figured that out.. Was doing it wrong earlier.

Thanks for your reply!

**Comment by jarrodek** [comment]
However I still strongly believe that this should be pointed out by parser / language server so the user won't get confused by meaningless (in this context) error message. It's quite hard to figure it out by your own without wasting time on the research.
